docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -0,0 +1,286 @@
+shards across the remaining nodes. However, you can use
+<<shard-allocation-awareness,shard allocation awareness>> to ensure shards for
+particular indices only move to specified nodes.
+
+[discrete]
+[[shard-size-guidelines]]
+=== Guidelines
+
+When applicable, we recommend you incorporate the following guidelines into your
+sharding strategy.
+
+[discrete]
+[[delete-indices-not-documents]]
+==== Delete indices, not documents
+
+Deleting individual documents often increases shard overhead. Deleted
+documents aren't directly removed from the file system. Instead, {es} marks the
+document as deleted on each related shard. The document will continue to use
+resources until it's removed in a segment <<index-modules-merge,merge>>.
+
+When possible, we recommend you delete entire indices instead. {es} can
+remove deleted indices directly from the file system, immediately freeing up
+resources.
+
+[discrete]
+[[use-ds-ilm-for-time-series]]
+==== For time series data, use data streams and {ilm-init}
+
+Data streams let you store time series data across multiple, time-based backing
+indices. You can then use <<index-lifecycle-management,{ilm} ({ilm-init})>>
+lifecycle policies to automatically manage these backing indices.
+
+One advantage of this setup is rollover, the process of creating a new write
+index when the current one meets a `max_age`, `max_docs`, or `max_size`
+threshold defined in your {ilm-init} policy. You can use these thresholds to
+group data into indices based on your retention period. You then can use
+{ilm-init} to automatically delete indices when they're no longer needed.
+
+{ilm-init} also makes it easy to change your sharding strategy over time:
+
+* Want to decrease the number of shards for new indices? +
+Change the <<index-number-of-shards,`index.number_of_shards`>> setting in the
+data stream's <<data-streams-change-mappings-and-settings,matching index
+template>>.
+
+* Want larger shards? +
+Increase the rollover threshold in your {ilm-init} policy.
+
+* Need to create smaller indices with shorter time periods? +
+Offset the increased shard count by deleting those older indices sooner.
+
+Each new backing index is an opportunity to further tune your strategy.
+
+[discrete]
+[[shard-size-recommendation]]
+==== Aim for shard sizes between 10GB and 50GB
+
+Shards larger than 50GB may make a cluster less likely to recover from failure.
+When a node fails, {es} rebalances the cluster's shards across its remaining
+nodes. Shards larger than 50GB can be harder to move across networks and may tax
+system resources.
+
+[discrete]
+[[shard-count-recommendation]]
+==== Aim for 20 shards or fewer per GB of heap space
+
+The number of shards a node can hold is proportional to the node's
+<<heap-size,heap space>>. For example, a node with a 30GB heap should have a
+maximum of 600 shards. If you find your nodes nearing more than 20 shards per
+GB, consider adding another node.
```

Comment:
Note for reviewers:

Note sure if this is still our guideline after heap improvements:
https://www.elastic.co/guide/en/elasticsearch/reference/7.7/release-highlights.html#_significant_reduction_of_heap_usage_of_segments